apis/projectcontour/v1/httpproxy.go

```diff
@@ -324,6 +328,12 @@ type RetryPolicy struct {
 	// PerTryTimeout specifies the timeout per retry attempt.
 	// Ignored if NumRetries is not supplied.
 	PerTryTimeout string `json:"perTryTimeout,omitempty"`
+	// RetryOn specifies the conditions on which to retry a request.
+	// +optional
+	RetryOn []RetryOn `json:"retryOn,omitempty"`
+	// RetriableStatusCodes specifies the HTTP status codes that should be retried.
+	// +optional
```

Comment:
This should document when the field is used (i.e. only when you set `RetryOn` to `retriable-status-codes`.